opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -832,6 +825,9 @@ func waitForVmState(d *schema.ResourceData, meta interface{}, state string) (int
 
 func generateVmXML(d *schema.ResourceData) (string, error) {
 
+	// Retrieve Name
+	name := d.Get("name").(string)
```

Comment:
name is not mandatory on OpenNebula. If it is not set is use the one generated by OpenNebula.
So in your case you will have VM with empty name if you don't provide it in the template neither than in the `tf` file.